src/Compilers/Core/Portable/Symbols/WellKnownMemberNames.cs

```diff
@@ -344,6 +344,16 @@ public static class WellKnownMemberNames
         public const string SliceMethodName = "Slice";
 
         // internal until we settle on this long-term
-        internal const string CloneMethodName = "<>Clone";
+        internal const string CloneMethodName = "<Clone>$";
```

Comment:
><Clone>$ [](start = 49, length = 8)

Is this consistent with other generated names? I think most generated names from the C# compiler are "<id1>id2" where "id1" is optional. The original "<>Clone" seemed consistent.

Reply:
> Is this consistent with other generated names? I think most generated names from the C# compiler are "<id1>id2" where "id1" is optional. The original "<>Clone" seemed consistent.

According to Tomas it is consistent, in fact he suggested making this change. See linked issue.

---
In reply to: [454453400](https://github.com/dotnet/roslyn/pull/45930#discussion_r454453400) [](ancestors = 454453400)

Reply:
@tmat, should the names be "<>Clone", "<>Program", and "<>Main" instead to be more consistent with other generated names?

Reply:
The implementation of [GeneratedName.TryParseGeneratedName](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Synthesized/GeneratedNames.cs,290) parses `[CS$]<[middle]>c[__[suffix]]` patterns, where `c` is a required character. 
In order for this method to recognize these names we would need to update the condition [here](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Synthesized/GeneratedNames.cs,317) to include `$` and add `$` to [GeneratedNameKind](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Synthesized/GeneratedNameKind.cs,ec2c9544450ff197,references) enum, but that can easily be done when needed.

Reply:
> we would need to update the condition here to include $

Why include "$"?

Reply:
> Why include "$" in the names?

To avoid any possible collisions with other generated names. For example we use "<Main>" in names for async main scenario. Is presence of the "$" going to cause any problems?